apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-description-links/src/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { createInterpolateElement } from '@wordpress/element';
+import { addFilter } from '@wordpress/hooks';
+import { ReactElement, JSXElementConstructor } from 'react';
+import blockLinks from './block-links-map';
+import DescriptionSupportLink from './inline-support-link';
+
+const addBlockSupportLinks = (
+	settings: {
+		[ key: string ]: string | ReactElement< string | JSXElementConstructor< any > >;
+	},
+	name: string
+) => {
+	// If block has a parent, use the parents name in the switch. This will apply the link to all nested blocks.
+	const isChild = settings[ 'parent' ];
+	const blockName = isChild ? settings[ 'parent' ].toString() : name;
+
+	if ( blockLinks.hasOwnProperty( blockName ) ) {
+		const descriptionWithLink = createInterpolateElement( '<InlineSupportLink />', {
+			InlineSupportLink: (
+				<DescriptionSupportLink
+					title={ String( settings.title ) }
+					url={ blockLinks[ blockName ].url }
+				>
+					{ settings.description }
+				</DescriptionSupportLink>
+			),
+		} );
+
+		settings.description = descriptionWithLink || settings.description;
```

Comment:
Can `descriptionWithLink` ever be undefined?

Reply:
I guess not. The only reason it would have is if there was no `blockName` which is already being checked in the if statement.